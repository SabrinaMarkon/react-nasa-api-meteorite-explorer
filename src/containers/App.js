import React, {Component} from 'react';
import Header from '../components/Header';
import Nav from '../components/Nav';
import SearchContainer from '../containers/SearchContainer';
import ResultsContainer from '../containers/ResultsContainer';
import Footer from '../components/Footer';
import axios from 'axios';

export default class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      searchresults: []
    }
    this.doSearch = this.doSearch.bind(this);
  }

  // componentDidMount() {
  //   fetch(API_URL)
  //     .then(res => res.json())
  //     .then(json => this.setState({ searchinput: json }))
  //     .catch(err => {
  //       console.log(err);
  //     });
  // }$select=lower(name), annual_salary

  doSearch = searchinput => {
    let API_URL = 'https://data.nasa.gov/resource/gh4g-9sfh.json?$where=upper(name)=upper(\'' + searchinput + '\')';
    axios.get(API_URL)
    .then(res => {
      const searchresults = res.data;
      this.setState({
        searchresults
      });
    })
    .catch(err => { console.log('Error fetching results: ' + err) })
  }

  render() {
    return (
      <>
        <Nav />
        <Header />
        <SearchContainer doSearch={this.doSearch} />
        <ResultsContainer searchresults={this.state.searchresults} />
        <Footer />  
      </>
    )
  }
}

