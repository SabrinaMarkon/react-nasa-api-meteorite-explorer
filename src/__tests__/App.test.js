import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

// components
import App from '../containers/App';

function setup() {
  const wrapper = shallow(<App />);
  //return { wrapper, props }
  return wrapper;
}

describe('Testing main App container', () => {
  it('Renders the main App component', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Should have text in the main div', () => {
    const wrapper = setup();
    const hastext = wrapper.find('div');
    expect(hastext).toBeDefined();
  });

  it('Should have the right text', () => {
    const wrapper = setup();
    const findit = wrapper.find('div').text();
    expect(findit).toEqual('Main App Component');
  });

  it('The component should match the snapshot', () => {
    const wrapper = setup();
    console.log(wrapper);
    expect(wrapper).toMatchSnapshot();
  });
});


