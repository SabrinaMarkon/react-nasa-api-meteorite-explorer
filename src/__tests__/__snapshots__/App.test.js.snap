// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing main App container Should render the page correctly 1`] = `ShallowWrapper {}`;

exports[`Testing main App container The component should match the snapshot 1`] = `
<div>
  Main App Component
</div>
`;
